### PR TITLE
feat: add chart with sample daily number of actions

### LINKED ISSUE #1: show sample daily number of actions chart

### PR DESCRIPTION
**Pull request by hasanagh**
adds a chart with sample daily number of actions

close #1

### CODE REVIEW

**Comment by hasanagh** [comment]
Hi all! So we are adding here a simple boilerplate chart to view the number of actions per day in the teacher's view for a space. Take a look, and if you have any suggestions/questions please let me know!

**Review by HadiKhai — APPROVED** [review]

**Review by RolandSall — APPROVED** [review]

File: src/components/modes/teacher/TeacherView.js [hunk 1]
```diff
@@ -152,7 +83,8 @@ export class TeacherView extends Component {
   });
 
   state = {
-    selectedStudent: null,
+    fromDate: moment.utc('04-01-2019'),
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js]: Do we not want this to be somewhat dynamic? Based at least on today's date or on the data.

File: src/components/modes/teacher/TeacherView.js [hunk 2]
```diff
@@ -196,45 +134,106 @@ export class TeacherView extends Component {
...
-                  </TableRow>
-                </TableHead>
-                <TableBody>
-                  {renderAppInstanceResources(appInstanceResources, this.props)}
-                </TableBody>
-              </Table>
+              <Typography
+                variant="h5"
+                color="inherit"
+                style={{ marginBottom: 10 }}
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js]: Factor out to class in `classes`.

File: src/components/modes/teacher/TeacherView.js [hunk 3]
```diff
@@ -196,45 +134,106 @@ export class TeacherView extends Component {
...
+                                {yFormatted}
+                              </span>
+                            </div>
+                          </>
+                        ),
+                      )}
+                    </div>
+                  );
+                }}
+                margin={{ top: 50, right: 110, bottom: 50, left: 60 }}
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js]: Can all this style stuff be factored out to `classes`? Or be dynamic?

File: src/config/types.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
+import PropTypes from 'prop-types';
+
```
Comment (juancarlosfarah) [inline src/config/types.js]: Maybe call this file `propTypes` to differentiate from other types?

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Looking good! Just some minor questions / comments / changes below.

**Comment by juancarlosfarah** [comment]
@hasanagh, did you have a look at these?

**Comment by hasanagh** [comment]
@juancarlosfarah  Hey! thanks for the review. yes will push my changes tonight.

File: src/components/modes/teacher/TeacherView.js (hunk 1 above)
Comment (hasanagh) [reply src/components/modes/teacher/TeacherView.js]: True, I wasn't sure which approach to take. Will default it to the boundaries of data for now and we can change it later.

File: src/components/modes/teacher/TeacherView.js [hunk 5]
```diff
@@ -161,10 +102,33 @@ export class TeacherView extends Component {
...
+
+  handleActionsDateRange = () => {
+    const { fromDate, toDate } = this.state;
+    const { actions } = this.props;
+    const rangedMoment = extendMoment(moment);
+    const start = rangedMoment.utc(fromDate);
+    const end = rangedMoment.utc(toDate);
+    const range = rangedMoment.range(start, end);
+    return Array.from(range.by('days')).map(day => ({
+      x: day.format('DD/MM/YYYY'),
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js]: Factor out `DD/MM/YYYY` to `DATE_FORMAT`.

File: src/components/modes/teacher/TeacherView.js [hunk 6]
```diff
@@ -161,10 +102,33 @@ export class TeacherView extends Component {
...
+      y: actions.find(({ date }) => date.isSame(day, 'days'))?.count ?? 0,
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js:131]: Are the `?.` and `??` operators being properly compiled when building the app?

File: src/components/modes/teacher/TeacherView.js [hunk 7]
```diff
@@ -196,45 +159,106 @@ export class TeacherView extends Component {
...
-                <TableBody>
-                  {renderAppInstanceResources(appInstanceResources, this.props)}
-                </TableBody>
-              </Table>
+              <Typography
+                variant="h5"
+                color="inherit"
+                className={classes.chartTitle}
+              >
+                {t('Daily number of actions')}
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js]: 'Number of Daily Actions'

File: src/components/modes/teacher/TeacherView.js [hunk 8]
```diff
@@ -196,45 +159,106 @@ export class TeacherView extends Component {
...
+                          serieColor,
+                          id,
+                        }) => (
+                          <>
+                            <div>
+                              Date:&nbsp;
+                              {xFormatted}
+                            </div>
+                            <div>
+                              Number Of actions:&nbsp;
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js]:
```javascript
{ `${t('Number of Actions')} ` }
```

File: src/components/modes/teacher/TeacherView.js [hunk 9]
```diff
@@ -196,45 +159,106 @@ export class TeacherView extends Component {
...
+                  precision: 'day',
+                }}
+                yScale={{
+                  type: 'linear',
+                  stacked: false,
+                }}
+                axisLeft={{
+                  orient: 'left',
+                  legendPosition: 'middle',
+                  legend: 'Number of actions',
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js]: `t('Number of Actions')`

File: src/components/modes/teacher/TeacherView.js [hunk 10]
```diff
@@ -196,45 +159,106 @@ export class TeacherView extends Component {
...
+                    >
+                      {slice.points.map(
+                        ({
+                          data: { xFormatted, yFormatted },
+                          serieColor,
+                          id,
+                        }) => (
+                          <>
+                            <div>
+                              Date:&nbsp;
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js]:
```javascript
{ `${t('Date')} ` }
```

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Thanks @hasanagh! Some minor changes and it will be good to go. Tagging @hagoptaminian, @HadiKhai and @RolandSall for their awareness.

File: src/components/modes/teacher/TeacherView.js (hunk 6 above)
Comment (hasanagh) [reply src/components/modes/teacher/TeacherView.js:131]: Yes, we should be fine. Babel `7.8.0` supports the two features. Anyway, I will double-check and get back to you.

Comment (juancarlosfarah) [reply src/components/modes/teacher/TeacherView.js:131]: 👍

**Review by juancarlosfarah — APPROVED** [review]

**Pull request merged**